### PR TITLE
modules: platform-independent module

### PR DESCRIPTION
**Pull request by NotAShelf**
Drops excessive reliance on Nixpkgs modules inside the module options, and prepares for further platform support.

We get rid of assertions done inside module options (bad idea, why did I do that?) and separate the module options into a "common" module to be imported. This opens the door to further platform support, most notably Darwin as discussed in #4.

Also makes using [ndg](https://github.com/feel-co/ndg) to document module *options* more possible, so we can offer module documentation once this is merged.

### CODE REVIEW

File: modules/common.nix [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            Whether to set the execute bit on the target file.
+          '';
+        };
+
+        clobber = mkOption {
+          type = bool;
+          default = clobberOpt;
+          defaultText = literalExpression "config.hjem.clobberByDefault";
+          description = ''
+            Whether to "clobber" existing target paths. While `true`, tmpfile rules will be constructed
```
Comment (eclairevoyant) [inline modules/common.nix:74]: mentioning `systemd-tmpfiles` seems linux-specific, maybe drop second sentence?

File: modules/nixos/default.nix [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+in {
+  imports = [
+    # This should be removed in the future, since 'homes' is a very vague
+    # namespace to occupy. Added 2024-12-27, remove 2025-01-27 to allow
+    # sufficient time to migrate.
+    (lib.mkRenamedOptionModule ["homes"] ["hjem" "users"])
+  ];
+
+  options.hjem = {
+    clobberByDefault = mkOption {
```
Comment (eclairevoyant) [inline modules/nixos/default.nix:47]: any reason to not put this in the common file?

Comment (NotAShelf) [reply modules/nixos/default.nix:47]: This is bit of a hack. We cannot pass the `config` into the submodule directly, unless you want to do something like `specialArgs = {osConfig = config;}`  just to make sure NixOS config is directly accessible in the submodule. Thus we

1. define `clobberByDefault` in the NixOS module
2. default value of `clobberFiles` (in common.nix) gets set to this value
3. the value of `clobber` in the files submodule gets set to the value of `clobberFiles`

Not very clean, but the only way I could figure out how to propagate the top-level module option to a submodule inside a submodule.

File: modules/common.nix (hunk 1 above)
Comment (NotAShelf) [reply modules/common.nix:74]: Sounds about right, systemd-tmpfiles won't be the only way to manage links in the future.

**Pull request merged**